File: terrasage-admin/src/types/species.ts
export type DifficultyLevel = "BEGINNER" | "INTERMEDIATE" | "ADVANCED";
export type CitesLevel = "APPENDIX_I" | "APPENDIX_II" | "APPENDIX_III";
export type SpeciesStatus = "DRAFT" | "PUBLISHED" | "ARCHIVED";
export type HandlingLevel = "EASY" | "MODERATE" | "DIFFICULT" | "EXPERT_ONLY";
export type GeneticPattern =
  | "DOMINANT" | "RECESSIVE" | "CO_DOMINANT" | "LINE_BRED"
  | "CULTIVAR" | "VARIEGATED" | "HYBRID" | "SPORT";
export type SpeciesCategory =
  | "REPTILE" | "AMPHIBIAN" | "FISH" | "INVERTEBRATE" | "MAMMAL" | "BIRD"
  | "SUCCULENT" | "CACTUS" | "ORCHID" | "FOLIAGE" | "CARNIVOROUS_PLANT" | "AQUATIC_PLANT" | "BONSAI"
  | "OTHER";

export interface SpeciesListItem {
  id: number;
  scientificName: string;
  commonNameKo: string;
  commonNameEn: string | null;
  thumbnailUrl: string | null;
  category: SpeciesCategory | null;
  difficultyLevel: DifficultyLevel;
  family: string;
  status: SpeciesStatus;
}

export interface AnimalCareGuide {
  enclosureType: string | null;
  enclosureSizeCm: string | null;
  substrate: string | null;
  tempHotZone: number | null;
  tempCoolZone: number | null;
  tempNight: number | null;
  humidityMin: number | null;
  humidityMax: number | null;
  uvbRequired: boolean;
  photoperiodHours: number | null;
  feedType: string | null;
  feedFrequency: string | null;
  supplements: string | null;
  handlingLevel: HandlingLevel | null;
  cohabitationNote: string | null;
}

export interface PlantCareGuide {
  potType: string | null;
  growingMedium: string | null;
  lightRequirement: string | null;
  lightHoursPerDay: number | null;
  tempMin: number | null;
  tempMax: number | null;
  humidityMin: number | null;
  humidityMax: number | null;
  wateringFrequency: string | null;
  wateringMethod: string | null;
  fertilizerType: string | null;
  fertilizerFrequency: string | null;
  repottingNote: string | null;
  pruningNote: string | null;
  overallNote: string | null;
}

export interface Variant {
  id: number;
  name: string;
  geneticPattern: GeneticPattern;
  description: string | null;
  imageUrl: string | null;
}

export interface SpeciesDetail {
  id: number;
  scientificName: string;
  commonNameKo: string;
  commonNameEn: string | null;
  kingdom: string;
  phylum: string;
  taxonomyClass: string;
  taxonomyOrder: string;
  family: string;
  genus: string;
  origin: string | null;
  habitat: string | null;
  lifespanCaptive: number | null;
  lifespanWild: number | null;
  avgSizeCm: number | null;
  avgWeightG: number | null;
  difficultyLevel: DifficultyLevel;
  category: SpeciesCategory | null;
  citesLevel: CitesLevel | null;
  legalStatusNote: string | null;
  thumbnailUrl: string | null;
  status: SpeciesStatus;
  animalCareGuide: AnimalCareGuide | null;
  plantCareGuide: PlantCareGuide | null;
  variants: Variant[];
}

export interface PageResponse<T> {
  content: T[];
  totalElements: number;
  totalPages: number;
  number: number;
  size: number;
}

export interface ApiResponse<T> {
  success: boolean;
  data: T | null;
  error: { code: string; message: string } | null;
}

File: terrasage-api/src/main/kotlin/com/terrasage/api/encyclopedia/entity/Species.kt
package com.terrasage.api.encyclopedia.entity

import jakarta.persistence.*
import java.time.LocalDateTime

// 생물 종 정보 (린네 분류 체계 기반)
@Entity
@Table(name = "species")
class Species(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long = 0,

    // 학명 (국제 표준, 종 식별의 유일 키)
    @Column(nullable = false, unique = true)
    var scientificName: String,

    @Column(nullable = false)
    var commonNameKo: String,

    var commonNameEn: String? = null,

    // 린네 분류 계층: 계 > 문 > 강 > 목 > 과 > 속
    // 'class'/'order'는 Kotlin 예약어이므로 taxonomy 접두어 사용
    @Column(nullable = false)
    var kingdom: String,

    @Column(nullable = false)
    var phylum: String,

    @Column(name = "taxonomy_class", nullable = false)
    var taxonomyClass: String,

    @Column(name = "taxonomy_order", nullable = false)
    var taxonomyOrder: String,

    @Column(nullable = false)
    var family: String,

    @Column(nullable = false)
    var genus: String,

    // 서식 정보
    var origin: String? = null,
    var habitat: String? = null,

    // 수명 (년 단위)
    var lifespanCaptive: Int? = null,
    var lifespanWild: Int? = null,

    // 평균 크기/무게
    var avgSizeCm: Double? = null,
    var avgWeightG: Double? = null,

    // 사육 난이도
    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    var difficultyLevel: DifficultyLevel,

    // CITES (멸종위기종 국제거래 협약) 등급
    @Enumerated(EnumType.STRING)
    var citesLevel: CitesLevel? = null,

    var legalStatusNote: String? = null,
    var thumbnailUrl: String? = null,

    // 취미 원예/사육 분류 카테고리 (taxonomy와 별개 — 검색/필터 UI용)
    @Enumerated(EnumType.STRING)
    var category: SpeciesCategory? = null,

    // 등록 → DRAFT, 관리자 승인 → PUBLISHED
    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    var status: SpeciesStatus = SpeciesStatus.DRAFT,

    @Column(nullable = false, updatable = false)
    val createdAt: LocalDateTime = LocalDateTime.now(),

    @Column(nullable = false)
    var updatedAt: LocalDateTime = LocalDateTime.now(),
)
